src/full/Agda/TypeChecking/Primitive.hs

```diff
@@ -1861,23 +1861,25 @@ gpi info name a b = do
   return $ El (piSort (getSort a) (Abs y (getSort b)))
               (Pi dom (Abs y b))
 
-hPi, nPi :: (MonadTCM tcm, MonadDebug tcm)
-         => String -> tcm Type -> tcm Type -> tcm Type
+hPi, nPi :: (MonadAddContext m, MonadDebug m)
+         => String -> m Type -> m Type -> m Type
 hPi = gpi $ setHiding Hidden defaultArgInfo
 nPi = gpi defaultArgInfo
 
-hPi', nPi' :: (MonadTCM tcm, MonadDebug tcm)
-           => String -> NamesT tcm Type -> (NamesT tcm Term -> NamesT tcm Type) -> NamesT tcm Type
+hPi', nPi' :: (MonadAddContext m, MonadDebug m)
+           => String -> NamesT m Type -> (NamesT m Term -> NamesT m Type) -> NamesT m Type
 hPi' s a b = hPi s a (bind' s b)
 nPi' s a b = nPi s a (bind' s b)
 
-pPi' :: (MonadTCM tcm, MonadDebug tcm)
-     => String -> NamesT tcm Term -> (NamesT tcm Term -> NamesT tcm Type) -> NamesT tcm Type
-pPi' n phi b = toFinitePi <$> nPi' n (elInf $ cl (liftTCM primIsOne) <@> phi) b
+pPi' :: (MonadAddContext m, HasBuiltins m, MonadDebug m)
+     => String -> NamesT m Term -> (NamesT m Term -> NamesT m Type) -> NamesT m Type
+pPi' n phi b = toFinitePi <$> nPi' n (elInf $ cl isOne <@> phi) b
  where
    toFinitePi :: Type -> Type
    toFinitePi (El s (Pi d b)) = El s $ Pi (setRelevance Irrelevant $ d { domFinite = True }) b
-   toFinitePi _               = __IMPOSSIBLE__
```

Comment:
Wait, ok apparently I did change something strange here. How did this not get caught by the cabal builds??